SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -163,24 +163,46 @@ Worker_RequestId USpatialSender::CreateEntity(USpatialActorChannel* Channel)
 		StablyNamedObjectRef = FUnrealObjectRef(0, 0, TempPath, OuterObjectRef);
 	}
 
+	// Classes can have a PIE prefix added to them. This needs to be removed
+	// to be able to properly look the class up in the schema database.
+	FString RemappedClassName = Class->GetPathName();
+	GEngine->NetworkRemapPath(NetDriver, RemappedClassName, false);
+
 	TArray<Worker_ComponentData> ComponentDatas;
 	ComponentDatas.Add(improbable::Position(improbable::Coordinates::FromFVector(Channel->GetActorSpatialPosition(Actor))).CreatePositionData());
 	ComponentDatas.Add(improbable::Metadata(Class->GetName()).CreateMetadataData());
 	ComponentDatas.Add(improbable::EntityAcl(ReadAcl, ComponentWriteAcl).CreateEntityAclData());
 	ComponentDatas.Add(improbable::Persistence().CreatePersistenceData());
 	ComponentDatas.Add(improbable::SpawnData(Actor).CreateSpawnDataData());
-	ComponentDatas.Add(improbable::UnrealMetadata(StablyNamedObjectRef, ClientWorkerAttribute, Class->GetPathName()).CreateUnrealMetadataData());
-	ComponentDatas.Add(improbable::Interest().CreateInterestData());
-	if (Actor->IsA<APlayerController>())
-	{
-		ComponentDatas.Add(improbable::Heartbeat().CreateHeartbeatData());
-	}
+	ComponentDatas.Add(improbable::UnrealMetadata(StablyNamedObjectRef, ClientWorkerAttribute, RemappedClassName).CreateUnrealMetadataData());
 
 	if (Class->HasAnySpatialClassFlags(SPATIALCLASS_Singleton))
 	{
 		ComponentDatas.Add(improbable::Singleton().CreateSingletonData());
 	}
 
+	// If the Actor was loaded rather than dynamically spawned, associate it with its owning sublevel.
+	if (Actor->HasAnyFlags(RF_WasLoaded))
+	{
+		if (uint32* ComponentId = ClassInfoManager->SchemaDatabase->LevelNameToComponentId.Find(Actor->GetTypedOuter<UWorld>()->GetName()))
+		{
+			ComponentDatas.Add(ComponentFactory::CreateEmptyComponentData(*ComponentId));
+		}
+		else
+		{
+			ComponentDatas.Add(ComponentFactory::CreateEmptyComponentData(SpatialConstants::NOT_SPAWNED_COMPONENT_ID));
```

Comment:
Should this ever happen?

Reply:
Maybe? Should we just crash here instead?

Reply:
I guess this would happen if you add a new sublevel and forget to regenerate schema?

Reply:
Yes it would happen then.

Reply:
Should we log the level name / actor name as a warning?